zap/src/main/java/org/zaproxy/zap/view/ZapSupportDialog.java

```diff
@@ -145,19 +154,32 @@ private JButton getBtnOpen() {
             btnOpen.setText(Constant.messages.getString("support.open.button"));
             btnOpen.setToolTipText(Constant.messages.getString("support.open.button.tooltip"));
             btnOpen.addActionListener(
-                    new java.awt.event.ActionListener() {
-                        @Override
-                        public void actionPerformed(java.awt.event.ActionEvent e) {
-                            try {
-                                Desktop.getDesktop().open(new File(Constant.getZapHome()));
-                            } catch (IOException e1) {
-                                LOGGER.error(
-                                        "An exception occurred while trying to have the OS open ZAP's Home Directory.",
-                                        e1);
-                            }
+                    e -> {
+                        try {
+                            Desktop.getDesktop().open(new File(Constant.getZapHome()));
+                        } catch (IOException e1) {
+                            LOGGER.error(
+                                    "An exception occurred while trying to have the OS open ZAP's Home Directory.",
+                                    e1);
                         }
                     });
         }
         return btnOpen;
     }
+
+    private JButton getBtnCopy() {
+        if (btnCopy == null) {
+            btnCopy = new JButton();
+            btnCopy.setText(Constant.messages.getString("support.copy.button"));
+            btnCopy.setToolTipText(Constant.messages.getString("support.copy.button.tooltip"));
+            btnCopy.addActionListener(
+                    e -> {
+                        StringSelection stringSelection =
+                                new StringSelection(ZapSupportUtils.getAll(true));
```

Comment:
e.g. `supportPanel.getInfo()` to ensure what's being shown is what's get copied (even in case of tweaks in the panel).